### ISSUE TITLE
We need to create Android Bundle .aab files directly from Animate (without Terminal)

### ISSUE THREAD

**Issue by wasserworks** [airsdk/Adobe-Runtime-Support#2460]
Hi @ajwfrost 

I don't know if you monitor the support emails at Harman, but I didn't get a reply about this:

Some of us have never used the Terminal and it's very daunting for a non developer. 

Would it be possible to add the option to create .aab files directly within Animate? 

Firstly, is it even possible, and secondly would you be able to work with adobe to get it done? 

It would need one more option button to publish an .aab, as well as separate Apks. 

If Android Studio is needed, we could link to that folder within the Air portal in Animate, and then click the .aab option. 

That would allow people from design and animation backgrounds like myself to continue making apps. Without the option to publish to a major App Store like Google Play really cuts down our options. 

Thanks,
Matt

**Comment by ajwfrost** [comment]
Hi

I don't personally monitor the adobe.support inbox, but one of my colleagues does and will tend to ping me with anything needing me to look at. I don't recall seeing your email, but I've looked back at my discussions with the Animate team and it's actually in mid-2021 that we had a conversation with them about the App Bundle option, and asset packs, etc...

Let me revive that discussion and see whether we can get them to provide this kind of option in the UI. We did add a workaround/hack option in the "adt.cfg" file, so if you set the below flag to 'true', Animate will think it's creating an APK file but will actually generate an app bundle:
```
# To override the creation of an Android APK file, and to instead create an Android App Bundle, the below
# flag can be uncommented. Note that the bundle may end up with an "apk" file extension
#CreateAndroidAppBundle=true
```

thanks

**Comment by wasserworks** [comment]
Yes, please do. That's quite a while ago! 

Thanks for the hack, I'll try it out. I hope I can just use a text editor to do that.

**Comment by josquinet** [comment]
yo lo hago con frecuencia y sigo estos pasos:

- en android studio instalar kos 2 útimos sdk platforms

- poner en archivo adt(en carpeta lib del air sdk) estos cambios:

quitar # en la línea:
CreateAndroidAppBundle=true

ruta sdk en android (creo que es doble barra para windows)
C:\\.....\\Local\Android\Sdk

ruta java
C:\\.....Android\\Android Studio\\jre


- en animate en configuracion de air sdk (el que sea) se ejecuta con CPU de 64 bits

- crear apk y cambiar apk a aab en el nombre

**Comment by wasserworks** [comment]
gracias por la lista de verificación!

**Comment by ajwfrost** [comment]
:-) 

On this part:
> (creo que es doble barra para windows)

Yes if you're using Windows the paths need to have either double-back-slash format, or single-forward-slash, for Java to interpret them properly. So:
`c:\\...\\local\\Android\\Sdk`
or
`c:/.../local/Android/Sdk`
Either should work..

Thanks

**Comment by wasserworks** [comment]
Hi guys, thanks for your help so far. I've just got round to installing Android Studio trying this hack. In theory it looks easy, but I get an error with the Java link. I'm using the same link that I got from Android studio, plus the **jre** bit at the end. 

**Question 1**

I'm on a MAC, and I see most of you are PC users. Here are the lines of code:

CreateAndroidAppBundle=true

AndroidPlatformSDK=/Users/mattssd/Library/Android/sdk

JAVA_HOME=/Users/mattssd/Library/Android/Android Studio/jre    

(I get an error with this. **Is JRE installed with Android Studio by default?**)

<img width="244" alt="Java error" src="https://user-images.githubusercontent.com/20012589/232134453-c944350c-f1a5-4af3-9753-c0113d4a02e7.png">

UseNativeCodesign=true    (should I use this on Mac?)

Question 2

I'm unsure if I'm supposed to install **all the previous Android Sdks** or just target 31 or 33? What do you recommend?

See screenshot:

<img width="843" alt="sdk versions" src="https://user-images.githubusercontent.com/20012589/232134524-f4ee34cf-1afa-44c7-a72d-db461cdc929e.png">


Thanks,

Matt

**Comment by wasserworks** [comment]
Right, so I might have solved the Java link, as Android Studio had given me the link in a different location:

JDK Location: /Applications/Android Studio.app/Contents/jbr/Contents/Home

The file published and I uploaded the .aab file to Google Play. But now I get a code signing error:

**"You uploaded an APK or Android App Bundle that is signed with a key that is also used to sign APKs that are delivered to users. Because you are enrolled in Play App Signing, you should sign your APK or Android App Bundle with a new key before you upload it."**

What to do next?!

**Comment by wasserworks** [comment]
To answer my last questions:  

1) I just made a new Cert on the Air Depoloyment page in Animate.
2) In Android Studio I kept the boxes ticked for 33 + 31. 

The aab uploaded fine, and this supported 18,000+ devices. 

If I should enable previous Android SDKs please let me know. 

This has gone much smoother that I thought!  

<img width="1242" alt="API levels" src="https://user-images.githubusercontent.com/20012589/232319138-7ecaf5e0-0103-4f9c-b503-9049674491f9.png">
